tests/pytests/functional/modules/test_mac_softwareupdate.py

```diff
@@ -0,0 +1,179 @@
+"""
+integration tests for mac_softwareupdate
+"""
+
+import pytest
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.skip_if_binaries_missing("softwareupdate"),
+    pytest.mark.skip_if_not_root,
+    pytest.mark.skip_unless_on_darwin,
+    pytest.mark.skip_initial_gh_actions_failure,
+]
+
+
+@pytest.fixture(scope="module")
+def softwareupdate(modules):
+    return modules.softwareupdate
+
+
+@pytest.fixture(scope="function", autouse=True)
+def _setup_teardown_vars(softwareupdate):
+    IGNORED_LIST = softwareupdate.list_ignored()
+
+    SCHEDULE = softwareupdate.schedule_enabled()
+
+    CATALOG = softwareupdate.get_catalog()
+
+    try:
+        yield IGNORED_LIST, SCHEDULE, CATALOG
+    finally:
+        if IGNORED_LIST:
+            for item in IGNORED_LIST:
+                softwareupdate.ignore(item)
+        else:
+            softwareupdate.reset_ignored()
+
+        softwareupdate.schedule_enable(SCHEDULE)
+
+        if CATALOG == "Default":
+            softwareupdate.reset_catalog()
+        else:
+            softwareupdate.set_catalog(CATALOG)
+
+
+def test_list_available(softwareupdate):
+    """
+    Test softwareupdate.list_available
+    """
+    # Can't predict what will be returned, so can only test that the return
+    # is the correct type, dict
+    ret = softwareupdate.list_available()
+    assert isinstance(ret, dict)
+
+
+@pytest.mark.skip(reason="Ignore removed from latest OS X.")
+def test_ignore(softwareupdate):
+    """
+    Test softwareupdate.ignore
+    Test softwareupdate.list_ignored
+    Test softwareupdate.reset_ignored
+    """
+    # Test reset_ignored
+    ret = softwareupdate.reset_ignored()
+    assert ret
+
+    ret = softwareupdate.list_ignored()
+    assert ret == []
+
+    # Test ignore
+    ret = softwareupdate.ignore("spongebob")
+    assert ret
+
+    ret = softwareupdate.ignore("squidward")
+    assert ret
+
+    # Test list_ignored and verify ignore
+    ret = softwareupdate.list_ignored()
+    assert "spongebob" in ret
+
+    ret = softwareupdate.list_ignored()
+    assert "squidward" in ret
+
+
```

Comment:
```suggestion
```
If it was removed, we remove the functionality and tests?!
No?

Reply:
Probably still works on older releases, should follow the normal deprecation path to remove.

Reply:
Ok.